### PR TITLE
host to ert interrupt

### PR DESCRIPTION
**Pull request by mamin506**
.

### CODE REVIEW

**Comment by xdavidz** [comment]
> Build Passed!

I an Min are collaborating code changes for ert handling. My code depends on this fix, thus his fix should go first. It is not fully ready yet.

**Comment by mamin506** [comment]
> I an Min are collaborating code changes for ert handling. My code depends on this fix, thus his fix should go first. It is not fully ready yet.

Test on Vega550.

File: src/runtime_src/core/edge/drm/zocl/sched_exec.c [hunk 1]
```diff
@@ -736,11 +684,32 @@ configure(struct sched_cmd *cmd)
...
+		cq_irq = zdev->ert->irq[ERT_CQ_IRQ];
+		ret = request_irq(cq_irq, sched_cq_isr, 0, "zocl_cq", zdev);
+		if (ret) {
+			DRM_WARN("Failed to initial CQ interrupt. "
+				 "Fall back to polling\n");
+			exec->cq_interrupt = 0;
+			exec->cq_thread = kthread_run(cq_check, zdev, name);
+		}
+	} else {
+		/* In CQ polling mode now */
```
Comment (xdavidz) [inline src/runtime_src/core/edge/drm/zocl/sched_exec.c]: why kthread_run cq_check again, it has been started in sched_init_exec

File: src/runtime_src/core/edge/drm/zocl/zocl_ert.c [hunk 2]
```diff
@@ -43,14 +188,18 @@ static int zocl_ert_probe(struct platform_device *pdev)
...
 
 	id = of_match_node(zocl_ert_of_match, pdev->dev.of_node);
 	ert_info(pdev, "Probing for %s", id->compatible);
 
 	ert = devm_kzalloc(&pdev->dev, sizeof(*ert), GFP_KERNEL);
 	ert->pdev = pdev;
-	ert->register_irq_handler = zocl_ert_irq_handler_register;
+
+	info = of_device_get_match_data(&pdev->dev);
+	if (info->model == ERT_MOD_RTL)
```
Comment (xdavidz) [inline src/runtime_src/core/edge/drm/zocl/zocl_ert.c]: NULL pointer for versal ERT, don't need this if?

**Comment by xdavidz** [comment]
Look ok to me. Passed test with versal code as well.

File: src/runtime_src/core/edge/drm/zocl/sched_exec.c [hunk 3]
```diff
@@ -736,11 +684,32 @@ configure(struct sched_cmd *cmd)
 		DRM_INFO("  cu_isr(%d)", exec->cu_isr);
 		DRM_INFO("  host_polling_mode(%d)", exec->polling_mode);
 		DRM_INFO("  cq_interrupt(%d)", exec->cq_interrupt);
-		setup_ert_hw(zdev);
+		zdev->ert->ops->config(zdev->ert, cfg);
 		exec->configured = 1;
 	}
 	write_unlock(&zdev->attr_rwlock);
 
+	/* Enable interrupt from host to PS when new commands are ready */
```
[the remark this answers is not in the document]
Comment (mamin506) [reply src/runtime_src/core/edge/drm/zocl/sched_exec.c]: I have similar thought. The scheduler just need to provide a call back. Let's refine it later.

**Pull request merged**